v2/concordium/protocol-level-tokens.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+package concordium.v2.plt;
+
+// This specifies the package we want to use for our generated Go code.
+// Has no effect on code generated on other languages.
+option go_package = "./pb";
+
+// This specifies the package we want to use for our generated Java classes.
+// Has no effect on code generated on other languages.
+option java_package = "com.concordium.grpc.v2.plt";
+
+// This specifies that separate .java files will be generated for each of the Java classes/enums/etc. generated for the top-level messages, services, and enumerations,
+// and the wrapper Java class generated for this .proto file won't contain any nested classes/enums/etc.
+// If not generating Java code, this option has no effect.
+option java_multiple_files = true;
+
+// This specifies the package we want to use for our generated C# classes.
+// Has no effect on code generated on other languages.
+option csharp_namespace = "Concordium.Grpc.V2.Plt";
+
+import "kernel.proto";
+
+// The unique symbol and identifier of a protocol level token.
+message TokenId {
+  // Unique identifier for the token, guaranteed to be distinct across the
+  // entire concordium blockchain.
+  string symbol = 1;
+}
+
+// A token module reference. This is always 32 bytes long.
+message TokenModuleRef {
+  bytes value = 1;
+}
+
+// PLT amount representation. The actual amount is computed as `digits *
+// 10^(-nr_of_decimals)`.
+message TokenAmount {
+  // The digits of the amount.
+  uint64 digits = 1;
+  // Number of decimals in the representation
+  uint32 nr_of_decimals = 2;
+}
+
+// Token state at the block level
+message TokenState {
+  // The token name. This should be unique, but uniqueness is enforced only by
+  // the chain governance, not by the consensus protocol.
+  string token_name = 1;
+  // The reference of the module implementing this token.
+  TokenModuleRef token_module_ref = 2;
+  // Account address of the issuer. The issuer is the holder of the nominated
+  // account which can perform token-governance operations.
+  concordium.v2.AccountAddress issuer = 3;
+  // Number of decimals in the decimal number representation of amounts.
+  uint32 nr_of_decimals = 4;
+  // A URL pointing to additional meta information about the token.
+  optional string token_meta_url = 5;
```

Comment:
In other cases (e.g. [CIS-2](https://proposals.concordium.com/CIS/cis-2.html#metadataurl)/[4](https://proposals.concordium.com/CIS/cis-4.html#metadataurl)), this is accompanied by an optional checksum to be able to verify the integrity of the underlying data. I don't see why we would differ here 🙂